Tests/Controller/RedirectToServiceControllerTest.php

```diff
@@ -11,70 +11,109 @@
 
 namespace HWI\Bundle\OAuthBundle\Tests\Controller;
 
+use HWI\Bundle\OAuthBundle\Controller\RedirectToServiceController;
+use HWI\Bundle\OAuthBundle\Security\OAuthUtils;
+use Symfony\Bundle\FrameworkBundle\Tests\TestCase;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
 
-class ConnectControllerRedirectToServiceActionTest extends AbstractConnectControllerTest
+class RedirectToServiceControllerTest extends TestCase
 {
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|OAuthUtils
+     */
+    private $oAuthUtils;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var array
+     */
+    private $firewallNames;
+
+    /**
+     * @var string
+     */
+    private $targetPathParameter;
+
     protected function setUp()
     {
         parent::setUp();
 
-        $this->container->setParameter('hwi_oauth.target_path_parameter', null);
-        $this->container->setParameter('hwi_oauth.use_referer', false);
-        $this->container->setParameter('hwi_oauth.failed_use_referer', false);
+        $this->oAuthUtils = $this->createMock(OAuthUtils::class);
+
+        $this->session = $this->createMock(SessionInterface::class);
+        $this->request = Request::create('/');
+        $this->request->setSession($this->session);
 
         $this->oAuthUtils->expects($this->any())
             ->method('getAuthorizationUrl')
             ->willReturn('http://domain.com/oauth/v2/auth')
         ;
+
+        $this->firewallNames = ['default'];
+        $this->targetPathParameter = 'target_path';
```

Comment:
Those two can be probably set directly on variable declaration as are not being changed.

Reply:
Done